lib/matplotlib/axes/_base.py

```diff
@@ -2541,23 +2541,20 @@ def handle_single_axis(scale, autoscaleon, shared_axes, interval,
             shared = shared_axes.get_siblings(self)
             dl = [ax.dataLim for ax in shared]
             # ignore non-finite data limits if good limits exist
-            finite_dl = [d for d in dl if np.isfinite(d).all()]
-            if len(finite_dl):
-                # if finite limits exist for at least one axis (and the
-                # other is infinite), restore the finite limits
-                x_finite = [d for d in dl
-                            if (np.isfinite(d.intervalx).all() and
-                                (d not in finite_dl))]
-                y_finite = [d for d in dl
-                            if (np.isfinite(d.intervaly).all() and
-                                (d not in finite_dl))]
-
-                dl = finite_dl
-                dl.extend(x_finite)
-                dl.extend(y_finite)
-
-            bb = mtransforms.BboxBase.union(dl)
-            x0, x1 = getattr(bb, interval)
+            # issue 18137: The previous code would allow one subplot with
+            # inifinite data limits to 'clobber' other subplots with finite
+            # data limits.
+            x_values = []
+            minpos_values = []
+            for d in dl:
+                x_values.extend(getattr(d, interval))
+                minpos_values.append(getattr(d, minpos))
+            x_values = np.sort(x_values).flatten()
+            finite_x_values = np.extract(np.isfinite(x_values), x_values)
+            if finite_x_values.size >= 1:
```

Comment:
Is there a good reason to pack the array, and then flatten it, and then get the min/max versus just doing min/max inside the loop?  This just seems like a lot of packing and unpacking for stuff that can be directly accessed.  I don't see that any of these variables are used below, but I may have missed it.  

e.g.
```python
minpos = np.Inf
x0 = np.Inf
x1 = -np.Inf
for ax in shared:
   for x in ax.dataLim.interval:
      if np.isfinite(x):
         x1 = np.max(x, x1)
         x0 = np.min(x, x0)
   if np.isfinite(ax.dataLim.minpos):
       minpos = np.min(ax.dataLim.minpos, minpos)
# then a check if still infinite...

```

Reply:
This code isn't mine, but I can take a look.

Reply:
This approach ends in several logical tests that in my view isn't as readable.